firefox-ios/Client/Frontend/Autofill/AutofillAccessoryView.swift

```diff
@@ -0,0 +1,112 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Common
+import Shared
+
+/// Custom UIBarButtonItem with an autofill accessory view.
+///
+/// This class provides a reusable and configurable autofill accessory view for use in navigation bars.
+///
+/// # Attributes
+/// - `accessoryImageViewSize`: Size of the accessory image view.
+/// - `accessoryButtonStackViewSpacing`: Spacing between the accessory image view and text label in the stack view.
+/// - `cornerRadius`: Corner radius for the accessory view.
+///
+/// # Properties
+/// - `accessoryImageViewTintColor`: Tint color for the accessory image view.
+/// - `backgroundColor`: Background color for the accessory view.
+///
+/// # Methods
+/// - `init(image:labelText:tappedAction:)`: Initializes the accessory view with an image, label, and optional tap action.
+/// - `tappedAccessoryButton()`: Handles the tap action on the accessory view.
+class AutofillAccessoryView: UIBarButtonItem {
+    // MARK: - Constants
+    private struct UX {
+        static let accessoryImageViewSize: CGFloat = 24
+        static let accessoryButtonStackViewSpacing: CGFloat = 2
+        static let cornerRadius: CGFloat = 4
+        static let fontSize: CGFloat = 16
+    }
+
+    // MARK: - Properties
+    private let accessoryImageView: UIImageView
+    private let useAccessoryTextLabel: UILabel
+    private let tappedAccessoryButtonAction: (() -> Void)?
+
+    /// Tint color for the accessory image view.
+    var accessoryImageViewTintColor: UIColor? {
+        get {
+            return accessoryImageView.tintColor
+        }
+        set {
+            accessoryImageView.tintColor = newValue
+        }
+    }
+
+    /// Background color for the accessory view.
+    var backgroundColor: UIColor? {
+        didSet {
+            updateBackgroundColor()
+        }
+    }
+
+    // MARK: - Initialization
+    /// Initializes the accessory view with an image, label, and optional tap action.
+    /// - Parameters:
+    ///   - image: The image for the accessory image view.
+    ///   - labelText: The text for the accessory view label.
+    ///   - tappedAction: The closure to be executed when the accessory view is tapped.
+    init(image: UIImage?, labelText: String, tappedAction: (() -> Void)? = nil) {
+        self.accessoryImageView = .build { imageView in
+            imageView.image = image?.withRenderingMode(.alwaysTemplate)
+            imageView.contentMode = .scaleAspectFit
+            imageView.accessibilityElementsHidden = true
+
+            NSLayoutConstraint.activate([
+                imageView.widthAnchor.constraint(equalToConstant: UX.accessoryImageViewSize),
+                imageView.heightAnchor.constraint(equalToConstant: UX.accessoryImageViewSize)
+            ])
+        }
+
+        self.useAccessoryTextLabel = .build { label in
+            label.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .title3, size: UX.fontSize, weight: .medium)
```

Comment:
I also see this styling in the Figma for "Use saved address": `iOS/Regular/Body`, where do I confirm that it's `title3 and `medium`?

![image](https://github.com/mozilla-mobile/firefox-ios/assets/6743397/7a28669e-950b-4742-9dd6-b1adbf17dfb8)

Reply:
![IMG_2682](https://github.com/mozilla-mobile/firefox-ios/assets/8919439/2e0b22c5-be1b-4ddb-beff-9da2232bb0e2)

@cyndichin its funny how I was looking at the same thing but in dark mode ha

Btw we should increase the hit area @jnrahme if possible and also add tint, I believe the button should have a tint when pressed as its current implementation I don't believe we have a tint